### PR TITLE
Feature/text widget

### PR DESCRIPTION
**Pull request by yahorm**

### CODE REVIEW

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/WidgetIndex.kt [hunk 1]
```diff
@@ -13,6 +13,7 @@ class WidgetIndex {
             JenkinsJobWidget::class.java.simpleName -> JenkinsJobWidget(vertx, config)
             SonarQubeWidget::class.java.simpleName -> SonarQubeWidget(vertx, config)
             ServiceCheckWidget::class.java.simpleName -> ServiceCheckWidget(vertx, config)
+            TextWidget::class.java.simpleName -> TextWidget(vertx, config)
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/WidgetIndex.kt]: In my opinion no backend is required for this widget.

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/TextWidget.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.cognifide.cogboard.widget.type
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/TextWidget.kt]: In my opinion no backend is required for this widget.

File: cogboard-webapp/src/components/widgets/dialogFields/TextInput.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ const TextInput = props => {
         shrink: true,
       }}
       margin="normal"
+      multiline={true}
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/widgets/dialogFields/TextInput.js]: Will this change look & fell of all other widget dialogs that use this field?

File: cogboard-webapp/src/components/widgets/dialogFields/index.js [hunk 4]
```diff
@@ -57,6 +58,25 @@ const dialogFields = {
...
+  Text: {
+    component: TextInput,
+    name: 'text',
+    label: 'Text'
+  },
+  TextSize: {
+    component: TextSizeInput,
+    name: 'textSize',
+    label: 'Text Size',
+    dropdownItems: {
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/widgets/dialogFields/index.js]: In my opinion this should be harcoded inside `TextSizeInput`

File: cogboard-webapp/src/components/widgets/index.js [hunk 5]
```diff
@@ -30,6 +31,11 @@ const widgetTypes = {
     component: ServiceCheckWidget,
     dialogFields: ['SchedulePeriod', 'URL', 'StatusCode']
   },
+  TextWidget: {
+    name: 'Text Widget',
+    component: TextWidget,
+    dialogFields: ['Text', 'TextSize']
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/widgets/index.js]: I cannot see text orientation property

File: cogboard-webapp/src/components/widgets/types/TextWidget.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+import Typography from '@material-ui/core/Typography';
+import {Requireable as string} from "prop-types";
+
+const TextWidget = props => {
+  const { text, textSize } = props;
+  return (
+      <Typography variant={textSize}>
+        <pre style={{ fontFamily: 'inherit' }}>
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/widgets/types/TextWidget.js]: is `pre` required?

File: cogboard-webapp/src/components/widgets/index.js (hunk 5 above)
Comment (yahorm) [reply cogboard-webapp/src/components/widgets/index.js]: will be added soon by @goeson00

File: cogboard-webapp/src/components/widgets/types/TextWidget.js (hunk 6 above)
Comment (yahorm) [reply cogboard-webapp/src/components/widgets/types/TextWidget.js]: same as above, will be changed soon by @goeson00

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/WidgetIndex.kt (hunk 1 above)
Comment (yahorm) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/WidgetIndex.kt]: `updateState` method used to update properties from edit window.

File: cogboard-webapp/src/components/widgets/dialogFields/TextInput.js (hunk 3 above)
Comment (yahorm) [reply cogboard-webapp/src/components/widgets/dialogFields/TextInput.js]: good point. it could be custom option just for text widget

File: cogboard-webapp/README.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
 The page will reload if you make edits.<br>
 You will also see any lint errors in the console.
 
-### `npm test`
+### `npm test`d
```
Comment (szymon-owczarzak) [inline cogboard-webapp/README.md]: revert this change

File: cogboard-webapp/src/components/widgets/dialogFields/index.js (hunk 4 above)
Comment (goeson00) [reply cogboard-webapp/src/components/widgets/dialogFields/index.js]: We have the same for EndpointInput. Leave it as it is.

File: cogboard-webapp/src/components/widgets/types/TextWidget.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  transform: rotate(-90deg);
+  -webkit-transform: rotate(-90deg);
+  -moz-transform: rotate(-90deg);
+  -ms-transform: rotate(-90deg);
+`;
+
+const TextWidget = props => {
+  const {text, textSize, isVertical} = props;
+
+  if (isVertical) {
```
Comment (goeson00) [inline cogboard-webapp/src/components/widgets/types/TextWidget.js]:
```suggestion
  const TextVariant = isVertical ? VerticalText : Typography;
  
  return (
  <TextVariant variant={textSize}>
  {text}
  </TextVariant>
  );
  
```

File: cogboard-webapp/src/components/widgets/types/TextWidget.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+import Typography from '@material-ui/core/Typography';
+import {bool, string} from "prop-types";
```
[the remark this answers is not in the document]
Comment (yahorm) [reply cogboard-webapp/src/components/widgets/types/TextWidget.js]: fixed

File: cogboard-webapp/src/components/widgets/dialogFields/CheckboxInput.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+const CheckboxInput = props => {
+  const {value} = props;
+
+  return (
+      <FormControlLabel
+          control={
+            <Checkbox
+                checked={value}
+                {...props}
```
[the remark this answers is not in the document]
Comment (yahorm) [reply cogboard-webapp/src/components/widgets/dialogFields/CheckboxInput.js]: fixed

File: cogboard-webapp/src/components/widgets/types/TextWidget.js (hunk 8 above)
Comment (yahorm) [reply cogboard-webapp/src/components/widgets/types/TextWidget.js]: fixed
